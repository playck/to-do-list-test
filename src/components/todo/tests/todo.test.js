import Todo from "../todo";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("Todo component", () => {
  const todo = { name: "todo", count: 3 };
  let TodoComponent;
  let handleIncrease;
  let handleDecrease;
  let handleRemoveTodo;

  beforeEach(() => {
    handleIncrease = jest.fn();
    handleDecrease = jest.fn();
    handleRemoveTodo = jest.fn();
    TodoComponent = (
      <Todo
        todo={todo}
        handleIncrease={handleIncrease}
        handleDecrease={handleDecrease}
        handleRemoveTodo={handleRemoveTodo}
      />
    );
  });

  it("renders", () => {
    const component = renderer.create(TodoComponent);
    expect(component.toJSON()).toMatchSnapshot();
  });

  describe("button Click", () => {
    beforeEach(() => {
      render(TodoComponent);
    });

    it('calls handleIncrease when click "increase" button', () => {
      const button = screen.getByTitle("increase");
      userEvent.click(button);
      expect(handleIncrease).toHaveBeenCalledWith(todo);
    });

    it('calls handleDecrease when click "decrease" button', () => {
      const button = screen.getByTitle("decrease");
      userEvent.click(button);
      expect(handleDecrease).toHaveBeenCalledWith(todo);
    });

    it('class handleRemove when click "remove" button', () => {
      const button = screen.getByTitle("remove");
      userEvent.click(button);
      expect(handleRemoveTodo).toHaveBeenCalledWith(todo);
    });
  });
});
