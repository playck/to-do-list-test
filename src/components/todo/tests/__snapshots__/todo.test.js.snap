// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component renders 1`] = `
<li
  className="todo"
>
  <span>
    todo
  </span>
  <span>
    3
  </span>
  <div
    className="button_wrapper"
  >
    <button
      onClick={[Function]}
      title="increase"
    >
      +
    </button>
    <button
      onClick={[Function]}
      title="decrease"
    >
      -
    </button>
    <button
      onClick={[Function]}
      title="remove"
    >
      X
    </button>
  </div>
</li>
`;
